File: Chalkable.Web/app/chlk/models/academicBenchmark/Document.js
REQUIRE('chlk.models.id.ABDocumentId');

NAMESPACE('chlk.models.academicBenchmark', function(){

    /**@class chlk.models.academicBenchmark.Document*/

    CLASS('Document', [
        chlk.models.id.ABDocumentId, 'id',
        String, 'code',
        String, 'description',

        function getName(){
            return this.getCode();
        }
    ]);
});

File: Chalkable.BusinessLogic/Model/AcademicBenchmark/Document.cs
using System;

namespace Chalkable.BusinessLogic.Model.AcademicBenchmark
{
    public class Document
    {
        public Guid Id { get; set; }
        public string Code { get; set; }
        public string Description { get; set; }

        public static Document Create(AcademicBenchmarkConnector.Models.Document doc)
        {
            return new Document
            {
                Id = doc.Id,
                Description = doc.Title,
                Code = doc.Code
            };
        }
    }
}
